backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -148,6 +166,29 @@ class VolunteerService implements IVolunteerService {
     return userVolunteerDTOs;
   }
 
+  async getCheckInsAndSchedules(
+    volunteerId: string,
+  ): Promise<(CheckInDTO | SchedulingDTO)[]> {
+    const checkIns = await (
+      await this.checkInService.getCheckInsByVolunteerId(volunteerId)
+    ).map((checkIn) => ({ ...checkIn, type: ShiftType.CHECKIN }));
+    const schedulings = await (
+      await this.schedulingService.getSchedulingsByVolunteerId(volunteerId)
+    ).map((scheduling) => ({ ...scheduling, type: ShiftType.SCHEDULING }));
+    const shifts = [...checkIns, ...schedulings].sort((a, b) => {
+      const date1 =
+        "startDate" in a
+          ? dayjs(a.startDate)
+          : getDateWithVolunteerTime(a.startTime, a.volunteerTime!);
```

Comment:
i think this `null` coalescing is okay? should always have a volunteer time if it goes through the volunteer schedulinig

Reply:
I made another comment about this where for some edge cases, it might be troublesome.

Reply:
would filtering out volunteerNeeded to be true fix this? why would a volunteer care about a scheduling that doesn't have a volunteerTime? doesn't that mean no volunteer is assigned?